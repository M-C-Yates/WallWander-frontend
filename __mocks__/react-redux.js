jest.mock('react-redux', () => {
	return {
		useDispatch() {
			return { dispatch: jest.fn() };
		},
		useSelector() {
			return {
				id: '12345',
				email: 'foo@bar.com',
				name: 'foobar',
				token: '123abc',
				nsfw: false
			};
		}
	};
});
