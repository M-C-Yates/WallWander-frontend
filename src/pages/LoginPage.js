import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { Link } from '@reach/router';

import { startLogin } from '../actions/auth';
import { UserForm } from '../components/UserForm';

const LoginPage = (props) => {
	const [email, setEmail] = useState('');
	const [password, setPassword] = useState('');

	const dispatch = useDispatch();

	const login = (e) => {
		e.preventDefault();
		const user = { email, password };
		dispatch(startLogin(user));

		props.navigate('/');
	};

	return (
		<React.Fragment>
			<UserForm
				button="Login"
				setEmail={setEmail}
				setPassword={setPassword}
				submit={login}
			/>
			<Link to="/register">Register</Link>
		</React.Fragment>
	);
};

export default LoginPage;
