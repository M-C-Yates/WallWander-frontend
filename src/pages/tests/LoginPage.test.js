import React from 'react';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

import { render, cleanup } from '@testing-library/react';
import { useDispatch } from 'react-redux';

import LoginPage from '../LoginPage';
// var MockAdapter = require('axios-mock-adapter');

jest.mock('react-redux', () => {
	return {
		useDispatch() {
			return { dispatch: jest.fn() };
		},
		useSelector() {
			return {
				id: '12345',
				email: 'foo@bar.com',
				name: 'foobar',
				token: '123abc'
			};
		}
	};
});

afterEach(cleanup);

describe('LoginPage', () => {
	it('matches snapshot', () => {
		const { asFragment } = render(<LoginPage />);
		expect(asFragment).toMatchSnapshot();
	});
});
