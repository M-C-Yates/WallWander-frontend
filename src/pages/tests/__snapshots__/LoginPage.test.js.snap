// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage matches snapshot 1`] = `[Function]`;
